vms/avm/network/network.go

```diff
@@ -53,37 +52,140 @@ type network struct {
 func New(
 	ctx *snow.Context,
 	parser txs.Parser,
-	manager executor.Manager,
+	txVerifier TxVerifier,
 	mempool mempool.Mempool,
 	appSender common.AppSender,
-) Network {
-	return &network{
-		AppHandler: common.NewNoOpAppHandler(ctx.Log),
+	registerer prometheus.Registerer,
+	txGossipHandlerID uint64,
+	maxValidatorSetStaleness time.Duration,
+	txGossipMaxGossipSize int,
+	txGossipPollSize int,
+	txGossipThrottlingPeriod time.Duration,
+	txGossipThrottlingLimit int,
+	maxExpectedElements uint64,
+	txGossipFalsePositiveProbability,
+	txGossipMaxFalsePositiveProbability float64,
+) (*Network, error) {
+	p2pNetwork, err := p2p.NewNetwork(ctx.Log, appSender, registerer, "p2p")
```

Comment:
nit: future backlog item might be finding a way to reduce the number of function invocations to initialize the p2p sdk.

Reply:
Yeah I found this pretty hard to reason about myself. Definitely room for improvement w.r.t. stitching all these objects together